### ISSUE TITLE
Change all catch(...) to catch (const std::exception &e)

### ISSUE THREAD

**Issue by kzangeli** [telefonicaid/fiware-orion#1512]
All catch(...) should be changed to use catch (const std::exception &e) instead.
Also, as we can probably get the error description using e.what(), the error handling should be enhanced.

**Effort: 1 man day**

**Comment by fgalan** [comment]
Do we have a reference to the C++ documentation where the `catch(...)` construct is explained to check that it is 100% equivalent to `catch (const std::exception &e)`. 

(In Java, for example, there are throwable object that are not exception, so a similar solution could have "gaps". I would like to ensure we don't have such gaps also in our case)

**Comment by kzangeli** [comment]
Here we have a discussion about this.
BTW, doesn't seem std::exception will catch all possible exceptions. Only STL exceptions ...

http://stackoverflow.com/questions/315948/c-catching-all-exceptions

**Comment by fgalan** [comment]
Thus, the optimal construct would be the follosing one:

```
try
{
  // Happy path code
}
catch (FooException &e)
{
  // Exception we know we can get
}
catch (std::exception &e)
{
  // Other uknown exception, but still in std::exception, with e.what(), etc.
}
catch (...)
{
  // Other unknwon expections
}
```

A bit complex, but fortunatelly the try/catch code is now located in few places (or refactor can be done to do it that way for some missing try/catch).

**Comment by fgalan** [comment]
Implemented in PR #1522
